### DISCUSSION TITLE
Specify the number of Epochs

### DISCUSSION THREAD

**Discussion by mchlsdrv** [MIC-DKFZ/nnUNet#490] [Q&A] [answered]
Hi, I would like to run the model just for 70 epochs, instead of 1000. Is there an option that can change the number of epochs in the command ? 

Thanks

**Comment by FabianIsensee** [comment, marked as answer]
You will need to create your own custom trainer for that. Simply derive from nnUNetTrainerV2 and set self.max_num_epochs=7 in init

**Reply by mchlsdrv** [reply]
Thanks, that worked!

**Reply by agatadefr** [reply]
> You will need to create your own custom trainer for that. Simply derive from nnUNetTrainerV2 and set self.max_num_epochs=7 in init
Hm... is there any example or tutorial how can I create my own custom trainer with different epochs value ?

**Comment by agatadefr** [comment]
Hm... is there any example or tutorial how can I create my own custom trainer with different epochs value ?

**Reply by FabianIsensee** [reply]
literally what is said above :upside_down_face: 

``` python
class nnUNetTrainerV2_500epochs(nnUNetTrainerV2):
    def __init__(self, plans_file, fold, output_folder=None, dataset_directory=None, batch_dice=True, stage=None,
                 unpack_data=True, deterministic=True, fp16=False):
        super().__init__(plans_file, fold, output_folder, dataset_directory, batch_dice, stage, unpack_data,
                         deterministic, fp16)
        self.max_num_epochs = 500
```

Place this in the network_training folder
